### PR TITLE
Add support for external java actions

### PR DESCRIPTION
**Pull request by Camork**
### Type of this PR 

- [ ] Bug fix
- [x] New feature or improvement
- [ ] Documentation
- [ ] Cleanup
- [ ] Test
- [ ] Build/Releng

### Issue(s)

The external java operation is useful for developers to implement complex business

### What does this PR do?

Add external java action support

### Screenshot/screencast of this PR

...
 

### Potential side effects

...

### How to test this PR?

- [ ] Frontend Unit tests
- [ ] Backend Unit tests
- [ ] Cypress : please specify
- [ ] Manual Test : please specify

### Checklist

- [x] I have read CONTRIBUTING carefully.
- [x] I have signed the [Contributor License Agreement](http://www.eclipse.org/legal/CLA.php).
- [x] All my commits are signed-off (`-s`) with my mail address of my Eclipse Account.
- [ ] I have covered my changes by unit tests or integration tests or manual tests.
- [ ] All tests pass.
- [ ] I have updated the documentation accordingly
- [ ] New React components are availables in storybook

### CODE REVIEW

File: backend/sirius-web-compatibility/src/main/java/org/eclipse/sirius/web/compat/api/IExternalJavaActionProvider.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package org.eclipse.sirius.web.compat.api;
```
Comment (sbegaudeau) [inline backend/sirius-web-compatibility/src/main/java/org/eclipse/sirius/web/compat/api/IExternalJavaActionProvider.java:1]: Do not forget the copyright on newly created files

File: backend/sirius-web-compatibility/src/main/java/org/eclipse/sirius/web/compat/utils/BeanUtil.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    @Override
+    public void setApplicationContext(@NonNull ApplicationContext applicationContext) throws BeansException {
+        context = applicationContext;
+    }
+
+    public static <T> T getBean(Class<T> beanClass) {
+        return context.getBean(beanClass);
+    }
+
+}
```
Comment (sbegaudeau) [inline backend/sirius-web-compatibility/src/main/java/org/eclipse/sirius/web/compat/utils/BeanUtil.java:28]: No need to use such class, use the constructor-based dependency injection from Spring

File: backend/sirius-web-emf/src/main/java/org/eclipse/sirius/web/emf/compatibility/ExternalJavaActionManager.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
+package org.eclipse.sirius.web.emf.compatibility;
```
Comment (sbegaudeau) [inline backend/sirius-web-emf/src/main/java/org/eclipse/sirius/web/emf/compatibility/ExternalJavaActionManager.java:1]: Do not forget the copyright on newly created files

File: backend/sirius-web-emf/src/main/java/org/eclipse/sirius/web/emf/compatibility/modeloperations/ExternalJavaActionOperationHandler.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
+package org.eclipse.sirius.web.emf.compatibility.modeloperations;
```
Comment (sbegaudeau) [inline backend/sirius-web-emf/src/main/java/org/eclipse/sirius/web/emf/compatibility/modeloperations/ExternalJavaActionOperationHandler.java:1]: Do not forget the copyright on newly created files

File: backend/sirius-web-emf/src/main/java/org/eclipse/sirius/web/emf/compatibility/ExternalJavaActionManager.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+            .map(Map::entrySet)
+            .flatMap(Collection::stream)
+            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    }
+
+    /**
+     * Returns the interpreter for the given expression.
+     *
+     * @param id the id of the wanted extension.
+     * @return the corresponding IJavaActionMenuItem.
```
Comment (sbegaudeau) [inline backend/sirius-web-emf/src/main/java/org/eclipse/sirius/web/emf/compatibility/ExternalJavaActionManager.java:41]: There is no `IJavaActionMenuItem` involved here

File: backend/sirius-web-emf/src/main/java/org/eclipse/sirius/web/emf/compatibility/modeloperations/ExternalJavaActionOperationHandler.java [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+                                              ChildModelOperationHandler childModelOperationHandler,
+                                              ExternalJavaAction externalJavaAction) {
+        this.interpreter = Objects.requireNonNull(interpreter);
+        this.childModelOperationHandler = Objects.requireNonNull(childModelOperationHandler);
+        this.externalJavaAction = Objects.requireNonNull(externalJavaAction);
+    }
+
+    @Override
+    public Status handle(Map<String, Object> variables) {
+        ExternalJavaActionManager actionManager = BeanUtil.getBean(ExternalJavaActionManager.class);
```
Comment (sbegaudeau) [inline backend/sirius-web-emf/src/main/java/org/eclipse/sirius/web/emf/compatibility/modeloperations/ExternalJavaActionOperationHandler.java:47]: Just ask for the ExternalJavaActionManager in the constructor

File: backend/sirius-web-emf/src/main/java/org/eclipse/sirius/web/emf/compatibility/modeloperations/ExternalJavaActionOperationHandler.java [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    private final ChildModelOperationHandler childModelOperationHandler;
+    private final ExternalJavaAction externalJavaAction;
+
+    public ExternalJavaActionOperationHandler(AQLInterpreter interpreter,
+                                              ChildModelOperationHandler childModelOperationHandler,
+                                              ExternalJavaAction externalJavaAction) {
+        this.interpreter = Objects.requireNonNull(interpreter);
+        this.childModelOperationHandler = Objects.requireNonNull(childModelOperationHandler);
+        this.externalJavaAction = Objects.requireNonNull(externalJavaAction);
+    }
```
Comment (sbegaudeau) [inline backend/sirius-web-emf/src/main/java/org/eclipse/sirius/web/emf/compatibility/modeloperations/ExternalJavaActionOperationHandler.java:43]: While this is not a big deal, this code is not formatted using our code formatter.

**Review by sbegaudeau — CHANGES_REQUESTED** [review]

**Comment by sbegaudeau** [comment]
Could you please open an issue for this feature request and update your commit message to match our conventions, thanks.

File: backend/sirius-web-emf/src/main/java/org/eclipse/sirius/web/emf/compatibility/modeloperations/ExternalJavaActionOperationHandler.java (hunk 6 above)
Comment (Camork) [reply backend/sirius-web-emf/src/main/java/org/eclipse/sirius/web/emf/compatibility/modeloperations/ExternalJavaActionOperationHandler.java:47]: The class ``ExternalJavaActionOperationHandler`` was initialized by 'new' and outside of the spring context , do we need to pass the ``ExternalJavaActionManager`` from the chain? That would make a lot of changes.

**Comment by Camork** [comment]
Hey @sbegaudeau , can you review again?

**Review by sbegaudeau — COMMENTED** [review]
I'll update the PR and fix the comment by myself to speed up its integration

**Comment by sbegaudeau** [comment]
I'll wait for the build and merge it

**Review by sbegaudeau — APPROVED** [review]

**Pull request merged**